main.cpp

```diff
@@ -189,6 +260,8 @@ DoubleType& DoubleType::operator/=(double d)
     return *this;
 }
 
+using IntFunctionPtr = void(*)(int&);
+
```

Comment:
This should be scoped to inside the IntType class.  it should not be a global alias.
- [ ] move alias.